### PR TITLE
How to configure admission plugins in kube-apiserver

### PR DESCRIPTION
**Pull request by mburke5678**
https://bugzilla.redhat.com/show_bug.cgi?id=1707381

### CODE REVIEW

**Comment by mburke5678** [comment]
@ravisantoshgudimetla Can you help with this PR? Any advice would be greatly appreciated. I am trying to close this by the end of the week.

File: modules/nodes-cluster-controllers-about.adoc [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+This admission controller overrides user-provided resource requirements in order to drive scheduling density. Requires the `autoscaling.openshift.io` API.
+
+RunOnceDuration::
+This admission controller specifies a maximum duration a run-once pod can run to completion.  Requires the `autoscaling.openshift.io` API.
+
+PodNodeConstraints::
+This admission controller restricts the set of node selectors that can appear on a pod unless they have pods/binding perm.  Requires the `autoscaling.openshift.io` API.
+
+OriginPodNodeEnvironment::
+This admission controller
```
Comment (mburke5678) [inline modules/nodes-cluster-controllers-about.adoc]: @sunilcio @ravisantoshgudimetla What does this admission controller do?

File: modules/nodes-cluster-admission-about.adoc [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+* NamespaceLifecycle
+* LimitRanger 
+* ServiceAccount
+* PersistentVolumeClaimResize
+* DefaultStorageClass
+* DefaultTolerationSeconds
+* MutatingAdmissionWebhook
+* ValidatingAdmissionWebhook
+* ResourceQuota
+* Priority
```
Comment (mburke5678) [inline modules/nodes-cluster-admission-about.adoc]: @ravisantoshgudimetla This list is from Kubernetes 1.13 docs. Is it accurate for OpenShift 4.1? Do the default admission controllers appear anywhere in code? The `kubeapiserver` object doesn't seem to list them.

File: modules/nodes-cluster-admission-configuring.adoc [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+kind: KubeAPIServer
+
+....
+
+unsupportedConfigOverrides:
+  admissionPluginConfig:
+    enabledPlugins:
+    - name: RunOnceDuration
+----
+
```
Comment (mburke5678) [inline modules/nodes-cluster-admission-configuring.adoc]: @sunilcio @ravisantoshgudimetla How do you handle different APIs? Create multiple kubeapiserver objects with different names?

File: modules/nodes-cluster-admission-about.adoc (hunk 2 above)
Comment (ravisantoshgudimetla) [reply modules/nodes-cluster-admission-about.adoc]: >  Do the default admission controllers appear anywhere in code?

AFAIK, you can find them when kube-apiserver starts.

Perhaps @openshift/sig-master can help.

**Comment by ravisantoshgudimetla** [comment]
/cc @openshift/sig-master 

@mburke5678 - I can help with some of the admission plugins but the way to configure them is owned by master team.

File: modules/nodes-cluster-admission-about.adoc [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+These are required for the proper functioning of the master. Modifying these lists is not
+recommended. Future versions of the product may use a different set of plug-ins and may change their ordering. If
+you do override the default list of plug-ins in the master configuration file,
+you are responsible for updating it to reflect requirements of newer versions of
+the {product-title} master.
+
+For the full list of admission controllers, see the 
+link:https://v1-13.docs.kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#what-does-each-admission-controller-do[Kubernetes documentation].
+
+AlwaysPullImages::
```
Comment (ravisantoshgudimetla) [inline modules/nodes-cluster-admission-about.adoc]: I believe this is deprecated. Some of these plugins are deprecated. PTAL - https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#what-does-each-admission-controller-do

Comment (mburke5678) [reply modules/nodes-cluster-admission-about.adoc]: @ravisantoshgudimetla I left the deprecated plug-ins in the list if they are in the [Admission Plugin Ownership spreadsheet](https://docs.google.com/spreadsheets/d/1UtdL44W2sLzJCAz8M1eBCTfFMvm9h7-3OT7w9GEzR90/edit#gid=0). I am trying to verify what is in that list.

**Comment by mburke5678** [comment]
@vikram-redhat @ravisantoshgudimetla @sunilcio 
@deads2k suggested closing this PR as unmerged because "we don't support configuring... the admission chain" 
Any thoughts?
 
https://coreos.slack.com/archives/CB48XQ4KZ/p1559160178087300

**Comment by vikram-redhat** [comment]
@mburke5678 happy with that. Please leave the note on the BZ as well. 

@sferich888, @aheslin  FYI

**Pull request closed without merging**